### PR TITLE
List related features sample: use correct identify workflow

### PR DESCRIPTION
**Pull request by jenmerritt**
### Description

Original request (CS 2032): Several samples use an identify workflow that is not best practice. We should be using `mapView.identifyLayersAsync(...)` when identifying across multiple layers or `mapView.identifyLayerAsync(...)` when we know which layer we need to run the identify on.

For Java the sample needing to be update is List Related Features. I've made what I think is the correct change, but @TADraeseke  if you could check this over first as per your original issue requirements, I can then get it reviewed for merging.

Thanks!

Checklist:
- [x] Set target branch to master (current release) or dev (next release)
- [x] Request a reviewer
- [x] Assign a reviewer
- [x] Tag reviewer in comment

### CODE REVIEW

**Review by TADraeseke — APPROVED** [review]
LGTM! Thanks @jenmerritt!

**Comment by jenmerritt** [comment]
@JonLavi @Rachael-E  very small update to list related features sample if you could both take a look, thanks!

File: feature_layers/list-related-features/src/main/java/com/esri/samples/list_related_features/ListRelatedFeaturesSample.java [hunk 1]
```diff
@@ -102,44 +99,43 @@ public void start(Stage stage) {
...
 
               try {
-                FeatureQueryResult result = selectFeatureQuery.get();
+                IdentifyLayerResult identifiedLayer = identifyLayerResult.get();
+                List<GeoElement> identifiedLayerGeoElements = identifiedLayer.getElements();
                 // get the first selected feature
-                Iterator<Feature> iterator = result.iterator();
-                if (iterator.hasNext()) {
-                  ArcGISFeature selectedFeature = (ArcGISFeature) iterator.next();
+                for (var geoElement : identifiedLayerGeoElements) {
```
Comment (JonLavi) [inline feature_layers/list-related-features/src/main/java/com/esri/samples/list_related_features/ListRelatedFeaturesSample.java]: For the sake of being extra clear which class we are dealing with here, we should write:
```suggestion
                for (GeoElement geoElement : identifiedLayerGeoElements) {
```

There's no harm in using `var` generally, but for samples we want to make sure our users know exactly which class they are dealing with at any point.

**Review by JonLavi — APPROVED** [review]
Nice update to this sample. Only thing I'm noticing is the use of `var`, which we shouldn't really do in samples. We want to be as verbose as possible. 
Approving so you can move on to a second/third review 👍

File: feature_layers/list-related-features/src/main/java/com/esri/samples/list_related_features/ListRelatedFeaturesSample.java (hunk 1 above)
Comment (Rachael-E) [reply feature_layers/list-related-features/src/main/java/com/esri/samples/list_related_features/ListRelatedFeaturesSample.java]: I agree with Jon here: when using `var`, it's typically to improve readability in the case of a super long class name, or multiple blocks of repeated class types. In this for loop, it's better to err on the side of readability with `GeoElement`.

Comment (jenmerritt) [reply feature_layers/list-related-features/src/main/java/com/esri/samples/list_related_features/ListRelatedFeaturesSample.java]: Understood, thanks both for the explanation. I've changed that @Rachael-E

**Review by Rachael-E — APPROVED** [review]

**Pull request merged**